scripts/update-translations.js

```diff
@@ -154,23 +158,94 @@ async function downloadVisualizationTranslations() {
             continue;
         }
 
         const file = `${pluginLocaleDir}/chart-translations.json`;
-        fs.writeFileSync(file, JSON.stringify(plugins[plugin], null, 2));
-
-        process.stdout.write(chalk`
-{green Updated visualization translations for plugin ${plugin}.}`);
+        await writeTranslationsIfGitClean(file, plugins[plugin]);
     }
 }
 
 async function go() {
     await downloadCoreTranslations();
     await downloadPluginTranslations();
     await downloadVisualizationTranslations();
+    await commitNewTranslations();
+}
+
+const gitFetchCache = new Set();
+const gitStatusCache = new Map();
+const gitNewChanges = new Map();
+
+async function writeTranslationsIfGitClean(file, translations) {
+    // read existing translations and compare
+    if (fs.existsSync(file)) {
+        const curTranslations = JSON.parse(await readFile(file, 'utf-8'));
+        if (isEqual(curTranslations, translations)) {
+            // no changes
+            return;
+        }
+    }
+    const repoPath = path.dirname(path.join(file, '../'));
+    const repoName = path.basename(repoPath);
+
+    // fetch latest origin
+    if (!gitFetchCache.has(repoPath)) {
+        process.stdout.write(chalk`
+{white New translations found for ${repoName}.}`);
+        // run git fetch
+        await exec('git fetch origin', { cwd: repoPath, shell: true });
+        gitFetchCache.add(repoPath);
+    }
+    // check that git status is empty (= clean repo)
+    if (!gitStatusCache.has(repoPath)) {
+        const { stdout: gitStatus } = await await exec('git status', {
+            cwd: repoPath,
+            shell: true
+        });
+        gitStatusCache.set(repoPath, gitStatus);
+    }
+    if (
+        !gitStatusCache.get(repoPath).includes('Your branch is up to date with') &&
+        !gitStatusCache.get(repoPath).includes('Your branch is ahead of')
+    ) {
+        process.stdout.write(chalk`
+{red ${repoName} is not clean, please run git pull before updating translations.}`);
+        return;
+    }
+
+    await writeFile(file, JSON.stringify(translations, null, 2));
+    if (!gitNewChanges.has(repoPath)) {
+        gitNewChanges.set(repoPath, []);
+        process.stdout.write(chalk`
+{green Updated translations for ${repoName}.}`);
+    }
+    gitNewChanges.get(repoPath).push(path.relative(repoPath, file));
+}
+
+async function commitNewTranslations() {
+    process.stdout.write('\n');
+    for (const [repoPath, files] of gitNewChanges.entries()) {
+        const repoName = path.basename(repoPath);
+        process.stdout.write(chalk`
+{white There are ${files.length} updated translation files in ${repoName}.}\n`);
+        try {
+            await question('  Do you want to commit them now? [y/N]  ');
+        } catch (answer) {
+            if (answer && answer.toLowerCase() === 'y') {
+                // commit changes
+                const cmd = `git commit ${files.join(' ')} -m "l10n: update translations"`;
```

Comment:
I think this should be safer (I haven't tested it though). It should prevent file names being parsed as options.

```suggestion
                const cmd = `git commit -m "l10n: update translations" -- ${files.join(' ')}`;
```